### Enoncé:
- a1: Expliquer la variable après rupture école en fonction des variables sexe, age, région, état civil, N-educatif, membre_famille, Revenu_famille, Revenu_mensuel, redoublant, prob_enseignant, Aide_école, Rq_comportement, Relation_parent, niveau educatif père, niveau éducatif mère, sign_école, endroit_fréquent, ph_famille, Ricule_prof et Ricule collègue.
- b1: Déterminer la qualité du modèle.
- a2: Expliquer la variable sensation après rupture école en fonction des variables sexe, age, région, état civil, N-educatif, membre_famille, Revenu_famille, Revenu_mensuel, redoublant, prob_enseignant, Aide_école, Rq_comportement, Relation_parent, niveau educatif père, niveau éducatif mère, sign_école, endroit_fréquent, ph_famille, Ricule_prof et Ricule collègue.
- b2: Déterminer la qualité du modèle.
- a3: Expliquer la variable violence en fonction des variables sexe, age, région, état civil, N-educatif, membre_famille, Revenu_famille, Revenu_mensuel, redoublant, prob_enseignant, Aide_école, Rq_comportement, Relation_parent, niveau educatif père, niveau éducatif mère, sign_école, endroit_fréquent, ph_famille, Ricule_prof et Ricule collègue.
- b3: Déterminer la qualité du modèle.

### Chargement des bibliothèques nécessaires et des données:

In [1]:
install.packages("readxl")
install.packages("nnet")
library(readxl)
library(nnet)

# Importer les données de l'exercice 1
df <- read_excel("data/exercice1.xlsx", sheet = "Feuil1")
head(df)

package 'readxl' successfully unpacked and MD5 sums checked


Warning message:
"cannot remove prior installation of package 'readxl'"
Warning message in file.copy(savedcopy, lib, recursive = TRUE):
"problem copying D:\Files\R-4.3.2\library\00LOCK\readxl\libs\x64\readxl.dll to D:\Files\R-4.3.2\library\readxl\libs\x64\readxl.dll: Permission denied"
Warning message:
"restored 'readxl'"



The downloaded binary packages are in
	C:\Users\medth\AppData\Local\Temp\Rtmp06qjUd\downloaded_packages
package 'nnet' successfully unpacked and MD5 sums checked


Warning message:
"cannot remove prior installation of package 'nnet'"
Warning message in file.copy(savedcopy, lib, recursive = TRUE):
"problem copying D:\Files\R-4.3.2\library\00LOCK\nnet\libs\x64\nnet.dll to D:\Files\R-4.3.2\library\nnet\libs\x64\nnet.dll: Permission denied"
Warning message:
"restored 'nnet'"



The downloaded binary packages are in
	C:\Users\medth\AppData\Local\Temp\Rtmp06qjUd\downloaded_packages


Warning message:
"package 'readxl' was built under R version 4.3.3"


ID sexe Region Age etat_civil N_educatif Membre_famille Revenu_famille
1 1  0    1      2   1          3           5             0             
2 2  0    1      4   2          3           5             0             
3 3  0    1      4   2          3          11             0             
4 5  0    1      3   2          3           4             0             
5 8  0    1      1   1          3           5             1             
6 9  0    1      4   3          3           4             1             
  Revenu_Mensuelle Redoublant ⋯ Niveau_educ_pere Niveau_educ_mere Sign_ecole
1   0              1          ⋯ 2                2                1         
2   0              1          ⋯ 2                1                1         
3 100              1          ⋯ 2                1                2         
4   0              1          ⋯ 1                1                1         
5   0              1          ⋯ 3                3                3         
6 100              0          ⋯ 2                3                4         
  Endroit_freq Ph_famille Ricule_Prof Ricule_collegue Sensation_apres_ecole
1 1            1          0           1               1                    
2 2            2          1           1               1                    
3 1            3          0           0               1                    
4 1            2          1           1               3                    
5 4            4          0           0               3                    
6 1            4          1           1               1                    
  Apres_repture_ecole Violence
1 1                   1       
2 1                   1       
3 2                   0       
4 3                   1       
5 2                   1       
6 2                   0

### Modélisation avec la variable <code>après_rupture_école</code>:

In [2]:
modele_a1 <- multinom(Apres_repture_ecole ~ sexe + Age +
                        Region + etat_civil + N_educatif +
                        Membre_famille + Revenu_famille +
                        Revenu_Mensuelle + Redoublant +
                        Prob_Enseignat + Aide_ecole +
                        Rq_Comprtement + Relat_Parent +
                        Niveau_educ_pere + Niveau_educ_mere +
                        Sign_ecole +
                        Endroit_freq + Ph_famille + Ricule_Prof +
                        Ricule_collegue, data = df)

summary(modele_a1)

# weights:  88 (63 variable)
initial  value 124.766493 
iter  10 value 78.622797
iter  20 value 63.239268
iter  30 value 55.258599
iter  40 value 49.846960
iter  50 value 47.101855
iter  60 value 43.719936
iter  70 value 43.481027
iter  80 value 43.465618
iter  90 value 43.465149
final  value 43.465144 
converged


Warning message in sqrt(diag(vc)):
"NaNs produced"


Call:
multinom(formula = Apres_repture_ecole ~ sexe + Age + Region + 
    etat_civil + N_educatif + Membre_famille + Revenu_famille + 
    Revenu_Mensuelle + Redoublant + Prob_Enseignat + Aide_ecole + 
    Rq_Comprtement + Relat_Parent + Niveau_educ_pere + Niveau_educ_mere + 
    Sign_ecole + Endroit_freq + Ph_famille + Ricule_Prof + Ricule_collegue, 
    data = df)

Coefficients:
  (Intercept)      sexe         Age       Region  etat_civil    N_educatif
2   -7.489772 245.87263  -0.4376716   -2.4084838    4.901879 -5.771419e-01
3  -10.469941  47.08085   0.7066992    0.6411367    5.941687  8.525498e-03
4 -611.468711  82.99164 208.3453744 -225.7999524 -128.520080 -1.605677e+02
  Membre_famille Revenu_famille Revenu_Mensuelle   Redoublant Prob_Enseignat
2      0.2134464      0.4469855       0.00356250    0.6462154      1.3745384
3      0.1245541      0.7613484       0.00124591   -0.2674273     -0.3408284
4     23.4909023    -32.9186956       0.42818081 -187.7576202    -38.2412999
  Aide_e

### Facteurs influençant l'abandon scolaire

**Genre :**
- Le genre joue un rôle significatif, avec un coefficient de **245,87** pour les garçons. Cette valeur élevée souligne l'impact important de cette variable, mais pourrait aussi indiquer une interaction avec d'autres facteurs liés.

**Âge :**
- Chaque année d'âge supplémentaire est associée à un effet modéré, représenté par un coefficient de **-0,44**. Cela suggère que l'augmentation de l'âge tend légèrement à diminuer les chances d'abandon scolaire, selon les contextes.

**Région :**
- La région géographique exerce une certaine influence, avec un coefficient négatif de **-2,41**. Cela reflète possiblement les inégalités régionales en matière d'accès à l'éducation.

**État matrimonial et niveau d'instruction des parents :**
- Un statut matrimonial stable agit comme un facteur protecteur contre l'abandon scolaire.
- Un niveau d'instruction plus élevé des parents, avec des coefficients de **0,41** pour le père et **0,99** pour la mère, est lié à une diminution du risque de décrochage.

**Revenu familial :**
- Un revenu plus élevé, avec des coefficients de **0,45** pour le revenu familial global et **0,0045** pour le revenu mensuel, réduit les chances d'abandon scolaire. Cela souligne l'importance des conditions financières pour la poursuite des études.

**Redoublement :**
- Le redoublement est corrélé à un risque accru d'abandon, avec un coefficient de **0,65**. Les élèves ayant redoublé semblent plus vulnérables à quitter le système scolaire.

**Qualité de l'enseignement :**
- La qualité de l'enseignement, mesurée par la probabilité d'avoir un enseignant compétent, influence fortement la continuité scolaire. Un coefficient de **1,37** montre qu'un enseignement de qualité est associé à un moindre risque d'abandon.

**Soutien scolaire :**
- La présence d’un accompagnement scolaire diminue les risques de décrochage, avec un coefficient de **0,52**.

**Comportement et relations familiales :**
- Les élèves présentant des comportements difficiles (coefficient de **-1,24**) ou ayant des relations conflictuelles avec leurs parents (coefficient de **-0,67**) ont un risque accru d'abandon scolaire.

**Problèmes familiaux :**
- Les conflits familiaux ou situations de tension augmentent également la probabilité d'abandon, avec un coefficient de **0,82**.

**Harcèlement scolaire :**
- Le harcèlement, qu'il soit causé par les enseignants ou les camarades, constitue un facteur de risque majeur. Les coefficients respectifs de **-1,19** et **0,82** reflètent ces influences négatives.

In [3]:
# Faire des prédictions
predictions <- predict(modele_a1, newdata = df)

# Matrice de confusion
confusion_matrix <- table(df$Apres_repture_ecole, predictions)
print(confusion_matrix)

# Calcul de l'exactitude
accuracy <- sum(diag(confusion_matrix)) / sum(confusion_matrix)
print(paste("Exactitude:", accuracy))

# Calcul de la précision, du rappel et du score F1
precision <- diag(confusion_matrix) / colSums(confusion_matrix)
recall <- diag(confusion_matrix) / rowSums(confusion_matrix)
f1_score <- 2 * (precision * recall) / (precision + recall)

# Afficher la précision, le rappel et le score F1 pour chaque classe
print(paste("Précision pour chaque classe:", precision))
print(paste("Rappel pour chaque classe:", recall))
print(paste("Score F1 pour chaque classe:", f1_score))

# Score F1 global
overall_f1_score <- mean(f1_score, na.rm = TRUE)
print(paste("Score F1 global:", overall_f1_score))

   predictions
     1  2  3  4
  1 29  0  0  0
  2  3 23  5  0
  3  2  5 12  1
  4  0  0  0 10
[1] "Exactitude: 0.822222222222222"


[1] "Précision pour chaque classe: 0.852941176470588"
[2] "Précision pour chaque classe: 0.821428571428571"
[3] "Précision pour chaque classe: 0.705882352941177"
[4] "Précision pour chaque classe: 0.909090909090909"
[1] "Rappel pour chaque classe: 1"                
[2] "Rappel pour chaque classe: 0.741935483870968"
[3] "Rappel pour chaque classe: 0.6"              
[4] "Rappel pour chaque classe: 1"                
[1] "Score F1 pour chaque classe: 0.920634920634921"
[2] "Score F1 pour chaque classe: 0.779661016949153"
[3] "Score F1 pour chaque classe: 0.648648648648649"
[4] "Score F1 pour chaque classe: 0.952380952380952"
[1] "Score F1 global: 0.825331384653418"


### Qualité du modèle

#### 1. Analyse de la matrice de confusion
La matrice de confusion montre le nombre de prédictions correctes et incorrectes pour chaque classe :

- **Classe 1** : Très bien prédite avec 29 prédictions correctes et aucune mauvaise classification.
- **Classe 2** : Performances acceptables, avec 23 prédictions correctes mais 5 instances mal classées comme Classe 3.
- **Classe 3** : Performances modérées, avec 12 prédictions correctes, mais 3 mal classées comme Classe 2 et 2 comme Classe 4.
- **Classe 4** : Performance parfaite, avec les 10 instances correctement classées.

#### 2. Exactitude (Accuracy)
- **Exactitude** : 82,22 %
    - Cela indique que 82,22 % de toutes les prédictions sont correctes. C’est une performance respectable, mais il y a une marge d’amélioration, notamment pour les Classes 2 et 3.

#### 3. Précision (Precision)
- **Classe 1** : 85,29 %
    - Bonne précision, ce qui signifie que la majorité des prédictions pour la Classe 1 sont correctes.
- **Classe 2** : 82,14 %
    - Une précision acceptable, bien qu’il y ait quelques prédictions incorrectes pour cette classe.
- **Classe 3** : 70,59 %
    - Précision plus faible, indiquant un taux élevé de faux positifs pour cette classe.
- **Classe 4** : 90,91 %
    - Excellente précision, avec très peu de faux positifs.

#### 4. Rappel (Recall)
- **Classe 1** : 100 %
    - Rappel parfait, toutes les instances de la Classe 1 ont été correctement identifiées.
- **Classe 2** : 74,19 %
    - Le modèle a légèrement du mal à détecter toutes les instances de la Classe 2.
- **Classe 3** : 60 %
    - Faible rappel, montrant que le modèle manque souvent des instances de la Classe 3.
- **Classe 4** : 100 %
    - Rappel parfait, aucune instance de la Classe 4 n’a été omise.

#### 5. Score F1
- **Classe 1** : 92,06 %
    - Excellent équilibre entre précision et rappel.
- **Classe 2** : 77,97 %
    - Performances acceptables, mais il y a une marge d’amélioration.
- **Classe 3** : 64,86 %
    - Le score F1 le plus faible, montrant des difficultés à équilibrer précision et rappel.
- **Classe 4** : 95,29 %
    - Excellent score F1, reflétant une très bonne performance pour cette classe.

#### 6. Score F1 global
- **Score F1 global** : 82,53 %
    - Ce score global montre que le modèle fonctionne raisonnablement bien pour toutes les classes, bien que la Classe 3 reste un point faible notable.

### Modélisation avec la variable <code>Sensation_apres_ecole</code>:

In [4]:
modele_a2 <- multinom(Sensation_apres_ecole ~ sexe + Age +
                        Region + etat_civil + N_educatif +
                        Membre_famille + Revenu_famille +
                        Revenu_Mensuelle + Redoublant +
                        Prob_Enseignat + Aide_ecole +
                        Rq_Comprtement + Relat_Parent +
                        Niveau_educ_pere + Niveau_educ_mere +
                        Sign_ecole +
                        Endroit_freq + Ph_famille +
                        Ricule_Prof + Ricule_collegue,
                      data = df)
summary(modele_a2)

# weights:  66 (42 variable)
initial  value 98.875106 
iter  10 value 69.921757
iter  20 value 63.545892
iter  30 value 62.811185
iter  40 value 62.710519
iter  50 value 62.693622
iter  60 value 62.691463
final  value 62.691445 
converged


Call:
multinom(formula = Sensation_apres_ecole ~ sexe + Age + Region + 
    etat_civil + N_educatif + Membre_famille + Revenu_famille + 
    Revenu_Mensuelle + Redoublant + Prob_Enseignat + Aide_ecole + 
    Rq_Comprtement + Relat_Parent + Niveau_educ_pere + Niveau_educ_mere + 
    Sign_ecole + Endroit_freq + Ph_famille + Ricule_Prof + Ricule_collegue, 
    data = df)

Coefficients:
  (Intercept)       sexe        Age    Region etat_civil N_educatif
2  -2.0092479 -15.288989 1.70302770  2.247275 -2.1451832 -1.1150235
3  -0.9491642   1.817137 0.05406014 -1.968661  0.4986185 -0.6205035
  Membre_famille Revenu_famille Revenu_Mensuelle Redoublant Prob_Enseignat
2    -0.06729289     -1.5735217     0.0023857860 -0.5856426     -0.1390584
3    -0.14471002     -0.3600626     0.0007416636  3.8445603      1.6831663
  Aide_ecole Rq_Comprtement Relat_Parent Niveau_educ_pere Niveau_educ_mere
2 -0.1679547     0.14149738  -0.33008611        -0.311379        0.9435157
3  0.3077777    -0.02068932   0.094

### Variables et leur interprétation

#### Sexe

**Coefficients :**
- Catégorie 2 : -15.29
- Catégorie 3 : 1.82

**Interprétation :**
- Le sexe a un impact négatif important sur la probabilité d'appartenir à la catégorie 2. Cela signifie que les individus de l'autre sexe (probablement féminin) sont beaucoup moins susceptibles d'être dans cette catégorie.
- Pour la catégorie 3, l'effet est positif, ce qui suggère que l'autre sexe augmente la probabilité d'appartenir à cette catégorie.

#### Âge

**Coefficients :**
- Catégorie 2 : 1.70
- Catégorie 3 : 0.05

**Interprétation :**
- L'âge a un effet positif sur la catégorie 2, indiquant que les personnes plus âgées ont une probabilité accrue d'appartenir à cette catégorie.
- Pour la catégorie 3, l'effet est très faible, suggérant que l'âge n'a pas d'influence significative.

#### Région

**Coefficients :**
- Catégorie 2 : 2.25
- Catégorie 3 : -1.97

**Interprétation :**
- Vivre dans une certaine région augmente fortement la probabilité d'appartenir à la catégorie 2.
- À l'inverse, cela réduit la probabilité d'appartenir à la catégorie 3.

#### État civil

**Coefficients :**
- Catégorie 2 : -2.14
- Catégorie 3 : 0.50

**Interprétation :**
- Un certain état civil (par exemple, célibataire) diminue fortement la probabilité d'être dans la catégorie 2.
- Pour la catégorie 3, l'effet est légèrement positif.

#### Niveau éducatif (N_educatif)

**Coefficients :**
- Catégorie 2 : -1.12
- Catégorie 3 : -0.62

**Interprétation :**
- Un niveau éducatif élevé réduit légèrement la probabilité d'appartenir aux catégories 2 et 3 par rapport à la catégorie de référence.

#### Membres de la famille (Membre_famille)

**Coefficients :**
- Catégorie 2 : -0.07
- Catégorie 3 : -0.14

**Interprétation :**
- Le fait d'avoir un certain nombre de membres dans la famille diminue légèrement la probabilité d'appartenir aux catégories 2 et 3.

#### Revenu familial (Revenu_famille)

**Coefficients :**
- Catégorie 2 : -1.57
- Catégorie 3 : -0.36

**Interprétation :**
- Un revenu familial plus élevé diminue la probabilité d'appartenir aux catégories 2 et 3, avec un impact plus marqué pour la catégorie 2.

#### Revenu mensuel (Revenu_Mensuelle)

**Coefficients :**
- Catégorie 2 : 0.002
- Catégorie 3 : 0.0007

**Interprétation :**
- Le revenu mensuel a un impact très faible (quasi-négligeable) sur la probabilité d'appartenir à ces catégories.

#### Redoublant

**Coefficients :**
- Catégorie 2 : -0.58
- Catégorie 3 : 3.84

**Interprétation :**
- Être un redoublant diminue légèrement la probabilité d'appartenir à la catégorie 2, mais augmente fortement celle d'appartenir à la catégorie 3.

#### Problèmes avec les enseignants (Prob_Enseignat)

**Coefficients :**
- Catégorie 2 : -0.14
- Catégorie 3 : 1.68

**Interprétation :**
- Les problèmes avec les enseignants diminuent légèrement la probabilité d'appartenir à la catégorie 2, mais augmentent fortement celle d'appartenir à la catégorie 3.

#### Aide à l'école (Aide_ecole)

**Coefficients :**
- Catégorie 2 : -0.17
- Catégorie 3 : 0.31

**Interprétation :**
- Recevoir une aide à l'école diminue légèrement la probabilité d'être dans la catégorie 2, mais augmente celle d'être dans la catégorie 3.

#### Remarques comportementales (Rq_Comprtement)

**Coefficients :**
- Catégorie 2 : 0.14
- Catégorie 3 : -0.02

**Interprétation :**
- Avoir des remarques comportementales augmente légèrement la probabilité d'appartenir à la catégorie 2, mais n'a presque aucun effet sur la catégorie 3.

#### Relation avec les parents (Relat_Parent)

**Coefficients :**
- Catégorie 2 : -0.33
- Catégorie 3 : 0.09

**Interprétation :**
- Une bonne relation avec les parents réduit la probabilité d'appartenir à la catégorie 2, mais n'a qu'un faible effet positif sur la catégorie 3.

#### Niveau éducatif des parents (Niveau_educ_pere et Niveau_educ_mere)

**Interprétation :**
- Les niveaux d'éducation des parents ont des effets variables :
    - Pour le père, un niveau élevé diminue la probabilité d'appartenir aux deux catégories.
    - Pour la mère, cela augmente fortement la probabilité pour les deux catégories.

#### Signes liés à l'école (Sign_ecole)

**Coefficients :**
- Catégorie 2 : -0.46
- Catégorie 3 : 0.27

**Interprétation :**
- Un signe spécifique à l'école diminue la probabilité d'appartenir à la catégorie 2, mais augmente légèrement celle de la catégorie 3.

#### Endroit fréquenté (Endroit_freq)

**Coefficients :**
- Catégorie 2 : 0.18
- Catégorie 3 : 0.10

**Interprétation :**
- Fréquenter certains endroits augmente légèrement la probabilité d'appartenir aux catégories 2 et 3.

#### Phénomènes familiaux (Ph_famille)

**Coefficients :**
- Catégorie 2 : 0.94
- Catégorie 3 : -0.61

**Interprétation :**
- Les phénomènes familiaux augmentent fortement la probabilité d'appartenir à la catégorie 2, mais diminuent celle de la catégorie 3.

#### Moqueries des professeurs et des collègues (Ricule_Prof et Ricule_collegue)

**Interprétation :**
- Ces variables mesurent les effets de moqueries des professeurs et des collègues :
- Les coefficients sont négatifs dans toutes les catégories, indiquant que ces comportements réduisent les probabilités d'appartenance.


In [5]:
predictions_a2 <- predict(modele_a2, newdata = df)

confusion_matrix_a2 <- table(df$Sensation_apres_ecole, predictions_a2)
print(confusion_matrix_a2)

accuracy_a2 <- sum(diag(confusion_matrix_a2)) / sum(confusion_matrix_a2)
print(paste("Exactitude:", accuracy_a2))

precision_a2 <- diag(confusion_matrix_a2) / colSums(confusion_matrix_a2)
recall_a2 <- diag(confusion_matrix_a2) / rowSums(confusion_matrix_a2)
f1_score_a2 <- 2 * (precision_a2 * recall_a2) / (precision_a2 + recall_a2)

print(paste("Précision pour chaque classe:", precision_a2))
print(paste("Rappel pour chaque classe:", recall_a2))
print(paste("Score F1 pour chaque classe:", f1_score_a2))

overall_f1_score_a2 <- mean(f1_score_a2, na.rm = TRUE)
print(paste("Score F1 global:", overall_f1_score_a2))

   predictions_a2
     1  2  3
  1 21  4  7
  2  3 23  6
  3  3  6 17
[1] "Exactitude: 0.677777777777778"
[1] "Précision pour chaque classe: 0.777777777777778"
[2] "Précision pour chaque classe: 0.696969696969697"
[3] "Précision pour chaque classe: 0.566666666666667"


[1] "Rappel pour chaque classe: 0.65625"          
[2] "Rappel pour chaque classe: 0.71875"          
[3] "Rappel pour chaque classe: 0.653846153846154"
[1] "Score F1 pour chaque classe: 0.711864406779661"
[2] "Score F1 pour chaque classe: 0.707692307692308"
[3] "Score F1 pour chaque classe: 0.607142857142857"
[1] "Score F1 global: 0.675566523871609"


### Qualité du modèle
#### 1. Matrice de confusion
- **Classe 1** : Correctement prédite 21 fois, incorrectement prédite 11 fois (4 fois comme classe 2, 7 fois comme classe 3).
- **Classe 2** : Correctement prédite 23 fois, incorrectement prédite 9 fois (3 fois comme classe 1, 6 fois comme classe 3).
- **Classe 3** : Correctement prédite 17 fois, incorrectement prédite 9 fois (3 fois comme classe 1, 6 fois comme classe 2).
#### 2. Exactitude
- **Exactitude** : 67.78%
    - Cela signifie que 67.78% des prédictions faites par le modèle sont correctes.
#### 3. Précision pour chaque classe
- **Classe 1** : 77.78% des prédictions étaient correctes.
- **Classe 2** : 69.70% des prédictions étaient correctes.
- **Classe 3** : 56.67% des prédictions étaient correctes.
#### 4. Rappel pour chaque classe
- **Classe 1** : Le modèle a capturé 65.63% des instances.
- **Classe 2** : Le modèle a capturé 71.88% des instances.
- **Classe 3** : Le modèle a capturé 65.38% des instances.
#### 5. Score F1 pour chaque classe
- **Classe 1** : Le score F1 pour la classe 1 est élevé, indiquant un bon équilibre entre précision et rappel.
- **Classe 2** : Le score F1 pour la classe 2 est également élevé, suggérant un bon équilibre entre précision et rappel.
- **Classe 3** : Le score F1 pour la classe 3 est plus faible, indiquant que le modèle pourrait être amélioré pour cette classe.
#### 6. Score F1 global
- **Score F1 global** : 0.68
    - Ce score indique un bon équilibre général entre précision et rappel pour toutes les classes.

> Le modèle montre une performance globale raisonnable avec une exactitude de 67.78% et un score F1 global de 0.68. Cependant, il existe des variations dans les performances pour chaque classe, avec des scores de précision et de rappel légèrement plus faibles pour la classe 3. Une optimisation supplémentaire pourrait être nécessaire pour améliorer la performance pour toutes les classes de manière équilibrée.

### Modélisation avec la variable <code>Violence</code>:

In [6]:
modele_a3 <- multinom(Violence ~ sexe + Age + Region + etat_civil +
                        N_educatif +
                        Membre_famille + Revenu_famille +
                        Revenu_Mensuelle + Redoublant +
                        Prob_Enseignat + Aide_ecole + Rq_Comprtement +
                        Relat_Parent +
                        Niveau_educ_pere + Niveau_educ_mere + Sign_ecole +
                        Endroit_freq + Ph_famille + Ricule_Prof +
                        Ricule_collegue,
                      data = df)
summary(modele_a3)

# weights:  22 (21 variable)
initial  value 62.383246 
iter  10 value 36.964839
iter  20 value 34.414180
final  value 34.343438 
converged


Call:
multinom(formula = Violence ~ sexe + Age + Region + etat_civil + 
    N_educatif + Membre_famille + Revenu_famille + Revenu_Mensuelle + 
    Redoublant + Prob_Enseignat + Aide_ecole + Rq_Comprtement + 
    Relat_Parent + Niveau_educ_pere + Niveau_educ_mere + Sign_ecole + 
    Endroit_freq + Ph_famille + Ricule_Prof + Ricule_collegue, 
    data = df)

Coefficients:
                       Values   Std. Err.
(Intercept)       6.068413312 3.713088928
sexe              1.524809931 2.123160842
Age              -1.055798304 0.533031905
Region            1.002753591 1.025679325
etat_civil        0.023586691 0.854480858
N_educatif       -0.673341017 0.624908213
Membre_famille   -0.061502670 0.193547317
Revenu_famille    1.282931321 0.974624342
Revenu_Mensuelle -0.004960213 0.002147526
Redoublant       -1.211504998 0.971903269
Prob_Enseignat   -2.011345947 0.982459715
Aide_ecole       -0.196828062 0.270754003
Rq_Comprtement    0.509204931 0.424193473
Relat_Parent      0.418167382 0.6716030

### Variables et leur interprétation pour la violence
#### Sexe
- **Valeur :** 1.525 (Erreur standard : 2.123)
- **Interprétation :** Être d'un certain sexe augmente les chances de violence par rapport à la catégorie de référence. La grande erreur standard indique une incertitude élevée dans cette estimation.
#### Âge
- **Valeur :** -1.056 (Erreur standard : 0.533)
- **Interprétation :** Chaque année d'âge supplémentaire réduit les chances de violence. L'effet est modéré et l'erreur standard indique une certaine incertitude.
#### Région
- **Valeur :** 1.003 (Erreur standard : 1.026)
- **Interprétation :** Vivre dans une certaine région augmente les chances de violence. L'erreur standard indique que cet effet est incertain.
#### État civil
- **Valeur :** 0.024 (Erreur standard : 0.854)
- **Interprétation :** L'état civil a un effet très limité sur les chances de violence, avec une grande incertitude.
#### Niveau éducatif
- **Valeur :** -0.673 (Erreur standard : 0.625)
- **Interprétation :** Un niveau d'éducation plus élevé réduit les chances de violence. L'effet est modéré avec une incertitude notable.
#### Membre de la famille
- **Valeur :** -0.062 (Erreur standard : 0.194)
- **Interprétation :** La présence de membres de la famille a un effet minime et incertain sur les chances de violence.
#### Revenu familial
- **Valeur :** 1.283 (Erreur standard : 0.975)
- **Interprétation :** Des revenus familiaux plus élevés augmentent les chances de violence, mais l'incertitude est élevée.
#### Revenu mensuel
- **Valeur :** -0.005 (Erreur standard : 0.002)
- **Interprétation :** Un revenu mensuel plus élevé réduit légèrement les chances de violence. L'effet est faible mais relativement précis.
#### Redoublant
- **Valeur :** -1.212 (Erreur standard : 0.972)
- **Interprétation :** Le fait de redoubler réduit les chances de violence, mais l'incertitude est élevée.
#### Problèmes avec les enseignants
- **Valeur :** -2.011 (Erreur standard : 0.982)
- **Interprétation :** Les problèmes avec les enseignants réduisent les chances de violence, avec une incertitude notable.
#### Aide scolaire
- **Valeur :** -0.197 (Erreur standard : 0.271)
- **Interprétation :** Le soutien scolaire a un effet minime et incertain sur les chances de violence.
#### Remarques sur le comportement
- **Valeur :** 0.509 (Erreur standard : 0.424)
- **Interprétation :** Les remarques sur le comportement augmentent les chances de violence, avec une incertitude notable.
#### Relation avec les parents
- **Valeur :** 0.418 (Erreur standard : 0.672)
- **Interprétation :** Une bonne relation avec les parents augmente les chances de violence, mais l'incertitude est élevée.
#### Niveau éducatif du père
- **Valeur :** -0.504 (Erreur standard : 0.734)
- **Interprétation :** Un niveau d'éducation plus élevé du père réduit les chances de violence, avec une incertitude notable.
#### Niveau éducatif de la mère
- **Valeur :** -0.204 (Erreur standard : 0.672)
- **Interprétation :** Un niveau d'éducation plus élevé de la mère réduit les chances de violence, avec une incertitude notable.
#### Signification de l'école
- **Valeur :** -0.094 (Erreur standard : 0.375)
- **Interprétation :** La perception de l'importance de l'école a un effet minime et incertain sur les chances de violence.
#### Endroit fréquenté
- **Valeur :** 0.392 (Erreur standard : 0.274)
- **Interprétation :** Les lieux fréquentés augmentent les chances de violence, avec une incertitude notable.
#### Phénomène familial
- **Valeur :** -1.159 (Erreur standard : 0.421)
- **Interprétation :** Les dynamiques familiales réduisent les chances de violence, avec une incertitude notable.
#### Moqueries des enseignants
- **Valeur :** 2.830 (Erreur standard : 1.229)
- **Interprétation :** Les moqueries des enseignants augmentent considérablement les chances de violence, mais l'incertitude est élevée.
#### Moqueries des collègues
- **Valeur :** 0.984 (Erreur standard : 0.898)
- **Interprétation :** Les moqueries des collègues augmentent les chances de violence, avec une incertitude notable.

In [7]:
predictions_a3 <- predict(modele_a3, newdata = df)

confusion_matrix_a3 <- table(df$Violence, predictions_a3)
print(confusion_matrix_a3)

accuracy_a3 <- sum(diag(confusion_matrix_a3)) / sum(confusion_matrix_a3)
print(paste("Exactitude:", accuracy_a3))

precision_a3 <- diag(confusion_matrix_a3) / colSums(confusion_matrix_a3)
recall_a3 <- diag(confusion_matrix_a3) / rowSums(confusion_matrix_a3)
f1_score_a3 <- 2 * (precision_a3 * recall_a3) / (precision_a3 + recall_a3)

print(paste("Précision pour chaque classe:", precision_a3))
print(paste("Rappel pour chaque classe:", recall_a3))
print(paste("Score F1 pour chaque classe:", f1_score_a3))

overall_f1_score_a3 <- mean(f1_score_a3, na.rm = TRUE)
print(paste("Score F1 global:", overall_f1_score_a3))

   predictions_a3
     0  1
  0 54  6
  1 12 18
[1] "Exactitude: 0.8"
[1] "Précision pour chaque classe: 0.818181818181818"
[2] "Précision pour chaque classe: 0.75"             
[1] "Rappel pour chaque classe: 0.9" "Rappel pour chaque classe: 0.6"
[1] "Score F1 pour chaque classe: 0.857142857142857"
[2] "Score F1 pour chaque classe: 0.666666666666667"
[1] "Score F1 global: 0.761904761904762"


### Qualité du modèle

#### 1. Matrice de confusion
- **Classe 0** : Correctement prédite 54 fois, incorrectement prédite 6 fois (comme classe 1).
- **Classe 1** : Correctement prédite 18 fois, incorrectement prédite 12 fois (comme classe 0).

#### 2. Exactitude
- **Exactitude** : 80%
    - Cela signifie que 80% des prédictions faites par le modèle sont correctes.

#### 3. Précision pour chaque classe
- **Classe 0** : 81.82%
    - Des prédictions faites pour la classe 0, 81.82% étaient correctes.
- **Classe 1** : 75%
    - Des prédictions faites pour la classe 1, 75% étaient correctes.

#### 4. Rappel pour chaque classe
- **Classe 0** : 90%
    - Le modèle a capturé 90% des instances de la classe 0.
- **Classe 1** : 60%
    - Le modèle a capturé 60% des instances de la classe 1.

#### 5. Score F1 pour chaque classe
- **Classe 0** : 0.86
    - Le score F1 pour la classe 0 est élevé, indiquant un bon équilibre entre précision et rappel.
- **Classe 1** : 0.67
    - Le score F1 pour la classe 1 est plus faible, indiquant que le modèle pourrait être amélioré pour cette classe.

#### 6. Score F1 global
- **Score F1 global** : 0.76
    - Ce score indique un bon équilibre général entre précision et rappel pour toutes les classes.

Le modèle montre une performance globale satisfaisante avec une exactitude de 80% et un score F1 global de 0.76. Cependant, il existe une variation dans les performances pour chaque classe, avec des scores de rappel et de précision plus faibles pour la classe 1. Une optimisation supplémentaire pourrait être nécessaire pour améliorer la performance pour toutes les classes de manière équilibrée.